cores/utils.py

```diff
@@ -1,2 +1,29 @@
+import json
+import jwt
+
+from django.http  import JsonResponse
+
+from my_settings  import SECRET_KEY, ALGORITHM
+from users.models import User
+
 def login_decorator():
-    pass
\ No newline at end of file
+    def wrapper(self, request, *args, **kwargs):
+        if 'Authorization' not in request.headers :
+            return JsonResponse ({'message' : 'UNAUTHORIZED'}, status=401)
+
+        access_token = request.headers.get('Authorization')
+
+        try:
+            payload      = jwt.decode(access_token, SECRET_KEY, ALGORITHM)
+            user         = User.objects.get(id=payload['user_id'])
+            request.user = user
+
+        except jwt.exceptions.DecodeError:
+            return JsonResponse({'MESSAGE': 'INVALID_TOKEN'}, status=401)
+
+        except User.DoesNotExist:
+            return JsonResponse({'MESSAGE': 'INVALID_USER'}, status=401)
+
+        return func(self, request,  *args, **kwargs)
```

Comment:
```suggestion
        try:
            access_token = request.headers['Authorization']
            payload      = jwt.decode(access_token, SECRET_KEY, ALGORITHM)
            user         = User.objects.get(id=payload['user_id'])
            request.user = user

        except jwt.exceptions.DecodeError:
            return JsonResponse({'MESSAGE': 'INVALID_TOKEN'}, status=401)

        except User.DoesNotExist:
            return JsonResponse({'MESSAGE': 'INVALID_USER'}, status=401)
            
        except KeyError:
            return JsonResponse({'MESSAGE': 'Key_Error'}, status=400)

        return func(self, request,  *args, **kwargs)
```
```python
user = User.objects.get(id=payload['user_id'])
```
해당 부분에서도 keyerror가 발생할 수 있으니 다음과 같이 keyerror을 추가하면 좋을 것 같습니다.

Reply:
수정 완료 후 push 완료 했습니다 !